spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -78,13 +73,10 @@ public ChannelBindingService(ChannelBindingServiceProperties channelBindingServi
 		ConsumerProperties consumerProperties =
 				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName);
 		if (binder instanceof ExtendedPropertiesBinder) {
-			ExtendedPropertiesBinder extendedPropertiesBinder = (ExtendedPropertiesBinder) binder;
-			Object extension = extendedPropertiesBinder.getExtendedConsumerProperties(inputChannelName);
-			ExtendedConsumerProperties extendedConsumerProperties = new ExtendedConsumerProperties(extension);
-			BeanUtils.copyProperties(consumerProperties, extendedConsumerProperties);
-			consumerProperties = extendedConsumerProperties;
+			consumerProperties = ProducerConsumerPropertiesUtil.getExtendedConsumerProperties((ExtendedPropertiesBinder) binder,
```

Comment:
The intention to move the producer/consumer properties extraction to a external class is to make a way for the `BinderAwareChannelResolver` to invoke them explicitly. Since both the `binder` isn't obtained differently in `BinderAwareChannelResolver` and `ChannelBindingService`'s bindProducer/Consumer, encapsulating inside the `ChannelBindingService` may not help BinderAwareChannelResolver. What do you think?